### ISSUE TITLE
Partial match not working in some instances with non-greedy capture

### ISSUE THREAD

**Issue by palle-k** [mrabarnett/mrab-regex#546]
I have the following pattern that includes a non-greedy capture:

```python
pattern = regex.compile(r"<thinking>.*?</thinking>")
```

When I try to do partial matching with it, I am observing some weird behaviour, where some cases that should match, do not:

```python
# The following cases match (expected):
pattern.match("<", partial=True)
pattern.match("<thinking", partial=True)
pattern.match("<thinking>", partial=True)

# The following cases do NOT match (unexpected)
pattern.match("<thinking>x", partial=True)
pattern.match("<thinking>xyz abc", partial=True)
pattern.match("<thinking>xyz abc foo", partial=True)

# The following cases do match again (expected)
pattern.match("<thinking>xyz abc foo ", partial=True)
pattern.match("<thinking>xyz abc foo bar", partial=True)
```

Are there any flags that I can use to prevent this behaviour?

By the way, it starts matching again when the length of the string after <thinking> becomes longer than the closing expression </thinking>

Environment:
```
$ pip3 show regex
Name: regex
Version: 2024.9.11
...

$ python3 --version
Python 3.12.6
```

**Comment by mrabarnett** [comment]
Fixed in regex 2024.11.6.

**Comment by palle-k** [comment]
Thank you for the quick resolution!
